### PR TITLE
Firestore TransactionOptions added, to specify maxAttempts

### PR DESCRIPTION
**Pull request by dconeybe**
This is a port of https://github.com/firebase/firebase-cpp-sdk/pull/966, https://github.com/firebase/firebase-ios-sdk/pull/9838 and https://github.com/firebase/firebase-android-sdk/pull/3664, which adds the new `TransactionOptions` class to Firestore, which can be used to specify the maximum number of attempts to run a transaction before giving up. Previously, this maximum was hardcoded to 5.

### CODE REVIEW

File: firestore/src/TransactionOptions.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  /// </summary>
+  public TransactionOptions() {
+  }
+
+  /// <summary>
+  /// The maximum number of attempts to commit, after which the transaction fails.
+  /// </summary>
+  ///
+  /// <remarks>
+  /// The default value is 5.
```
[the remark this answers is not in the document]
Comment (dconeybe) [reply firestore/src/TransactionOptions.cs]: Done.

**Comment by dconeybe** [comment]
https://github.com/firebase/firebase-unity-sdk/actions/runs/2591114395 triggered to build and run the full suite of integration tests, this time I've temporarily cherry-picked an Android NDK fix (https://github.com/firebase/firebase-unity-sdk/pull/318/commits/5a934e1c45b306c1600caa1c395df165c4e9bc28) to hopefully fix the build issues in the `build-2019-macos-latest-Android` action when running the integration tests.

**Comment by dconeybe** [comment]
https://github.com/firebase/firebase-unity-sdk/actions/runs/2594580720 started after merging in the Android fix from the main branch.

**Comment by jonsimantov** [comment]
Hmm, the Android build is still failing. @sunmou99 any idea if this is related?

**Comment by sunmou99** [comment]
> Hmm, the Android build is still failing. @sunmou99 any idea if this is related?

The unity installation tool seems unstable. I will add retry in the tool itself.

**Comment by dconeybe** [comment]
Build workflow: https://github.com/firebase/firebase-unity-sdk/actions/runs/2624659988

Integration tests: https://github.com/firebase/firebase-unity-sdk/actions/runs/2625079199

**Pull request merged**